qmi/instruments/santec/tsl_570.py

```diff
@@ -0,0 +1,891 @@
+                f"({self._frequency_range.min}{unit} - {self._frequency_range.max}{unit})"
+            )
+
+        self._write_and_check_errors(f":WAV:FREQ:SWE:STOP {frequency}")
+
+    @rpc_method
+    def get_sweep_stop_frequency(self) -> float:
+        """Get the sweep stop frequency. Default unit is THz.
+
+        Returns:
+            frequency: The instrument frequency in nanometers.
+        """
+        frequency = self._ask_float(":WAV:FREQ:SWE:STOP?")
+        return frequency
+
+    @rpc_method
+    def set_sweep_mode(self, mode: int) -> None:
+        """Set sweep mode. Possible modes are:
+        0 - Step sweep mode and One way
+        1 - Continuous sweep mode and One way
+        2 - Step sweep mode and Two way
+        3 - Continuous sweep mode and Two way
+
+        Parameters:
+            mode: Integer in range [0, 3].
+        """
+        if mode not in range(4):
+            raise ValueError(f"Sweep mode value {mode} not in valid modes range [0-3].")
+
+        self._write_and_check_errors(f":WAV:SWE:MOD {mode}")
+
+    @rpc_method
+    def get_sweep_mode(self) -> int:
+        """Get sweep mode.
+
+        Returns:
+            mode: Integer in range [0, 3].
+        """
+        return self._ask_int(":WAV:SWE:MOD?")
+
+    @rpc_method
+    def set_sweep_speed(self, speed: int) -> None:
+        """Set sweep speed. Possible speed values are 1,2,5,10,20,50,100,200nm/s.
+
+        Parameters:
+            speed: speed in nm/s.
+        """
+        if speed not in [1, 2, 5, 10, 20, 50, 100, 200]:
+            raise ValueError(f"Sweep speed {speed} not in valid values [1,2,5,10,20,50,100,200]nm/s.")
+
+        self._write_and_check_errors(f":WAV:SWE:SPE {speed}")
+
+    @rpc_method
+    def get_sweep_speed(self) -> float:
+        """Get sweep speed.
+
+        Returns:
+            speed: Speed in range [1, 200]nm/s.
+        """
+        return self._ask_float(":WAV:SWE:SPE?")
+
+    @rpc_method
+    def set_sweep_dwell(self, dwell: float) -> None:
+        """Set _stepped_ sweep dwell. Possible dwell range is 0-999.9s in 0.1nm steps. It does not take into account
+        delay in one-way sweeps to return to the start frequency.
+
+        Parameters:
+            dwell: Dwell between sweep steps in range [0, 1000[s.
+        """
+        dec = 1
+        if dwell < 0.0 or dwell >= 1000.0:
+            raise ValueError(f"Sweep dwell {dwell} not in valid range [0-1000[s.")
+
+        self._write_and_check_errors(f":WAV:SWE:DWEL {dwell:.{dec}f}")
+
+    @rpc_method
+    def get_sweep_dwell(self) -> float:
+        """Get dwell between sweep steps.
+
+        Returns:
+            dwell: dwell in range [0, 1000[s.
+        """
+        return self._ask_float(":WAV:SWE:DWEL?")
+
+    @rpc_method
+    def set_sweep_delay(self, delay: float) -> None:
+        """Set _continuous_ sweep delay. Possible delay range is 0-999.9s in 0.1nm steps. It does not take into account
```

Comment:
what are the underscores for?

Reply:
To emphasize that this is for _continous_ mode, not for _step_ mode.